assignment3/word_search.py

```diff
@@ -0,0 +1,93 @@
+#! /usr/bin/python3.5
+
+import dictionary
+ALL_WORDS = set()
+WORD_DICT = None
+CHAR_2D_ARRAY = None
+
+def findAllWords(row, col, char2dArray, wordDict):
```

Comment:
So you've decided that it is best to try to construct all possible words from the grid and cull all the prefixes that are not found in the dictionary. That sounds entirely reasonable.

How would it compare to doing it the other way around - taking all the words in the dictionary and seeing if they can be found in the grid. I'm not really looking for thorough analysis, just interested in how you made that choice?